frontend/webadmin/modules/frontend/src/main/java/org/ovirt/engine/ui/frontend/server/dashboard/DashboardDataServlet.java

```diff
@@ -362,6 +369,20 @@ private GlobalUtilization lookupGlobalUtilization() throws DashboardDataExceptio
         return utilization;
     }
 
+    private String lookupEngineGrafanaBaseUrl(EngineLocalConfig config) {
+        try {
+            String grafanaFqdnVal = config.getEngineGrafanaFqdn();
```

Comment:
Both properties seem mandatory - the getter `getProperty(key, allowMissing)` makes it explicit by passing `allowMissing == false`. Perhaps we should not catch the resulting IllegalArgumentException?

Reply:
It's the same logic as on the welcome page (via branding):
https://github.com/oVirt/ovirt-engine/blob/9c84398f28aee78cd6fb8aeae07e2fa1f7d2b247/backend/manager/modules/branding/src/main/java/org/ovirt/engine/core/branding/BrandingManager.java#L383-L406

If the keys are not configured, it is ok.  The engine can be configured without grafana.